chainer/link_hook.py

```diff
@@ -147,7 +147,7 @@ def __exit__(self, *_):
         link_hooks[self.name].deleted(None)
         del link_hooks[self.name]
 
-    def added(self, link: tp.Optional['chainer.link.Link']) -> None:
```

Comment:
Errors occurred a string (`'chainer.link.Link'`) was evaluated as an ordinal class in Python3.5. 🤔 
I avoided this error by surrounding a whole annotation with whitespace.